File: ruby/lib/steam/packets/a2a_ack_packet.rb
# This code is free software; you can redistribute it and/or modify it under the
# terms of the new BSD License.
#
# Copyright (c) 2008-2010, Sebastian Staudt

require 'steam/packets/steam_packet'

# The A2A_ACK_Packet class represents the response to a A2A_PING
# request send to the server.
class A2A_ACK_Packet < SteamPacket

  # Creates a A2A_ACK response object based on the data received.
  def initialize(data)
    if data != "\0" && data != "00000000000000\0"
      raise Exception.new('Wrong formatted A2A_ACK packet.')
    end

    super SteamPacket::A2A_ACK_HEADER, data
  end

end

File: php/lib/steam/packets/A2A_ACK_Packet.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Packets
 */

require_once STEAM_CONDENSER_PATH . 'steam/packets/SteamPacket.php';

/**
 * @package Steam Condenser (PHP)
 * @subpackage Packets
 */
class A2A_ACK_Packet extends SteamPacket
{
	/**
	 * @param byte[] $contentData
	 */
	public function __construct($contentData)
	{
		if($contentData != "\0" && $contentData != "00000000000000\0")
		{
			throw new Exception("Wrong formatted A2A_ACK packet.");
		}
		parent::__construct(SteamPacket::A2A_ACK_HEADER, $contentData);
	}
}
?>